tools/system_libs.py

```diff
@@ -327,12 +327,6 @@ class Dummy:
       for dep in value:
         shared.Settings.EXPORTED_FUNCTIONS.append('_' + dep)
 
-  all_needed = set()
```

Comment:
Tangentially related. I found out that `all_needed` was unused while investigating the `test_sizeof` failure, and I bucketed its removal here under "general cleanup and probably not worth a PR on its own," similar to the other refactors.